src/_pytest/config/__init__.py

```diff
@@ -1078,12 +1085,33 @@ def _checkversion(self):
                     )
                 )
 
-    def _validatekeys(self):
+    def _validate_keys(self) -> None:
         for key in sorted(self._get_unknown_ini_keys()):
-            message = "Unknown config ini key: {}\n".format(key)
-            if self.known_args_namespace.strict_config:
-                fail(message, pytrace=False)
-            sys.stderr.write("WARNING: {}".format(message))
+            self._emit_warning_or_fail("Unknown config ini key: {}\n".format(key))
+
+    def _validate_plugins(self) -> None:
+        required_plugins = sorted(self.getini("required_plugins"))
+        if not required_plugins:
+            return
```

Comment:
I'd omit these two lines as they're not really necessary.

```suggestion
```

Reply:
This was intentional. The point is to not fetch plugin info if it’s not necessary. The function is useless if there are no required plugins, so we exit

Reply:
I don't think this optimization is worth it, the code below doesn't do anything expensive. I think it is better to avoid special cases when they are not necessary. But it's OK if you want to keep it.

Reply:
The call to get dist info is expensive ( as per @nicoddemus’ earlier comment )

Reply:
Yeah I was under that impression, but [I was wrong](https://github.com/pytest-dev/pluggy/blob/f17598a7b422a8a7566a258d2f79fae5fde6b78a/src/pluggy/manager.py#L286-L289), sorry for not doing the proper research before.

Regardless I think the change is harmless and particularly I would prefer to bail out early. 😁

Reply:
+1 on bailing early. It’s unnecessary otherwise. We’re not writing high frequency trading software but at the same time we should do basic optimizations

Reply:
The main thing is that the condition makes it look like a Falsey `required_plugins` has a special meaning. If it is removed, the reader doesn't need to figure out that it doesn't.

But just in general, if I see e.g.  `sum` implemented like this:

```py
def sum(xs):
    if not xs:
        return 0
    s = 0
    for x in xs:
        s += x
    return s
```

that seems redundant...

Reply:
I mean, technically speaking a falsely `required_plugins` means to the function and the reader of the function "nothing important to do here now, just carry on"

In this case we also don't have to assemble plugin_dist_names `plugin_dist_names = [dist.project_name for _, dist in plugin_info]` which we still would if we don't bail early